server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/builder/BaseStarTreeBuilder.java

```diff
@@ -141,6 +139,37 @@ public List<MetricAggregatorInfo> generateMetricAggregatorInfos(MapperService ma
         return metricAggregatorInfos;
     }
 
+    /**
+     * Get star tree document from the segment for the current docId with the dimensionReaders and metricReaders
+     */
+    protected StarTreeDocument getStarTreeDocument(
+        int currentDocId,
+        SequentialDocValuesIterator[] dimensionReaders,
+        List<SequentialDocValuesIterator> metricReaders
```

Comment:
any reason why metricReaders is a list and dimensionsReaders an array?

Reply:
I think its because of how we can initialize the dimensionReaders straight away as we know the dimensions length.

For metric readers, we have to go through each metric and each stat and its easier if we use list.